### ISSUE TITLE
Gama Breaker 2: grid from big tiff image will hang/auto-force-close Gama

### ISSUE THREAD

**Issue by hqnghi88** [gama-platform/gama#2485]
**Describe the bug**
In line with the series "how to break gama" , and actually it was an issue from an user, i load a big tiff (in term of row x column, not file size), gama's memory reach 2gb (the default) and hanged, sometime it auto-close. This issue has low priority because it purely depends on data and we can increase memory, just to tell a case which provide dirty workspace relating with some recent issues.

**To Reproduce**
1. Here is the model
[GamaBreaker.zip](https://github.com/gama-platform/gama/files/2237857/GamaBreaker.zip)

2. Just launch the experiment GamaBreaker
3. Gama is not responding or is closed

**Expected behavior**
This issue is tested with memory increase and it works. So expected behavior is a memory listener(?!?) to break the process and tell user about memory limitation, show them a workaround like "increase heap space in preference" .
 
**Desktop (please complete the following information):**
 - OS: win 10
 - GAMA version: git,release,ci
 - Java version 1.8

**Comment by AlexisDrogoul** [comment]
Unfortunately, there's not much we can do about this. OutOfMemoryError are complex to track (it can happen anywhere and in any thread). I've pushed a commit that tries to handle them in the most usual places, but it doesn't work 100% of the time (remember that even opening a dialog to tell that there isn't enough memory can be impossible !). 
Regarding tracking them before they happen, it's almost impossible as well: the memory used is not very accurate and cannot take into account the GC (which might free memory).

**Comment by AlexisDrogoul** [comment]
I have added an automated polling of the memory available that can be configured in the preferences. It will report a warning in the standard error view. Added also the possibility to stop the experiment or directly exit GAMA when OutOfMemory errors happen. Both of these two possibilities should hopefully prevent such issues. 


<img width="686" alt="image" src="https://user-images.githubusercontent.com/579256/43363513-a781ae0c-9330-11e8-9aa7-b83fd14e297d.png">
